File: tests/typesafety/python/logged_steps_misuse.py
# expect: fails
# title: A Logged total spent as a number, rejected by the checker
# misuse-of: logged_steps
from dataclasses import dataclass


@dataclass(frozen=True)
class Logged[T]:
    value: T
    log: tuple[str, ...]


def audited_total(n: int) -> Logged[int]:
    return Logged(n + n // 10, (f"taxed {n}",))


def net(n: int) -> int:
    return audited_total(n) - 45

File: tests/typesafety/typescript/logged_steps_misuse.ts
// expect: fails

type Logged<T> = { readonly value: T; readonly log: readonly string[] };

function auditedTotal(n: number): Logged<number> {
  return { value: n + Math.floor(n / 10), log: [`taxed ${n}`] };
}

export function net(n: number): number {
  return auditedTotal(n) - 45;
}
